lib/client.js

```diff
@@ -678,7 +678,7 @@ MqttClient.prototype._cleanUp = function (forced, done) {
   }
 
   if (forced) {
-    this.stream.destroy()
+    this.stream.end()
```

Comment:
I would prefer this to stay as `destroy()`, or are you seeing test failures because of it? `destroy()` ensures no more data is sent in the socket.